provider/bidengine/order.go

```diff
@@ -360,6 +360,13 @@ loop:
 
 	// cancel reservation
 	if !won {
+		if clusterch != nil {
+			result := <-clusterch
+			clusterch = nil
+			if result.Error() == nil {
+				reservation = result.Value().(ctypes.Reservation)
+			}
+		}
```

Comment:
good find.  should this be done after `cancel()`?

Reply:
The code was arranged like that at one point and caused all sorts of weird errors. We'd cancel the context then wait on the results and log a bunch of errors (instead of things being cleaned up)